exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/eu/FederationGatewayClient.java

```diff
@@ -0,0 +1,70 @@
+package fi.thl.covid19.exposurenotification.eu;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+public class FederationGatewayClient {
+
+    private final RestTemplate restTemplate;
+    private final String gatewayUrl;
+
+    public FederationGatewayClient(
+            @Qualifier("federationGatewayRestTemplate") RestTemplate restTemplate,
+            @Value("${covid19.federation-gateway.url-template}") String gatewayUrl
+    ) {
+        this.restTemplate = restTemplate;
+        this.gatewayUrl = gatewayUrl;
+    }
+
+    public int upload(String batchTag, String batchSignature, byte[] batchData) {
```

Comment:
I would have guessed that interpreting status-codes would be FederationGatewayClient's responsibility, meaning that those could be processed here and the return value could be just OK/NOK (boolean). If it's the service's responsibility (which might make sense) then maybe it would be nicer to give out http statuscode object instead of an int.

Reply:
207 is partial success. I haven't check what this means, yet. But let's check this when it's ready.